### PR TITLE
fix: use the result of AccessControlBaseController#handle

### PR DESCRIPTION
**Pull request by artsiomkorzun**

### CODE REVIEW

**Comment by Maxim-Gadalov** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-397.nightly-test.deltixhub.io
>E2E tests status: success

File: src/main/java/com/epam/aidial/core/Proxy.java [hunk 1]
```diff
@@ -101,8 +102,19 @@ public void handle(HttpServerRequest request) {
     }
 
     private void handleError(Throwable error, HttpServerRequest request) {
-        log.error("Can't handle request", error);
-        respond(request, HttpStatus.INTERNAL_SERVER_ERROR);
+        if (!request.response().ended()) {
+            HttpStatus status = HttpStatus.INTERNAL_SERVER_ERROR;
+            String message = null;
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/Proxy.java:107]: this may lead to NPE

File: src/main/java/com/epam/aidial/core/ProxyContext.java [hunk 2]
```diff
@@ -124,7 +124,8 @@ public Future<Void> respond(HttpStatus status, String body) {
                     body.length() > LOG_MAX_ERROR_LENGTH ? body.substring(0, LOG_MAX_ERROR_LENGTH) : body);
         }
 
-        return response.setStatusCode(status.getCode()).end(body);
+        response.setStatusCode(status.getCode()).end(body);
+        return Future.succeededFuture();
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/ProxyContext.java]: why so? why do we need to succeed future before response body actually written

File: src/main/java/com/epam/aidial/core/controller/DownloadFileController.java [hunk 3]
```diff
@@ -29,40 +29,38 @@ protected Future<?> handle(ResourceDescription resource, boolean hasWriteAccess)
...
+                        return Future.succeededFuture();
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }).onFailure(error -> context.respond(HttpStatus.INTERNAL_SERVER_ERROR,
+                        "Failed to fetch file with path %s/%s".formatted(resource.getBucketName(),
+                                resource.getOriginalPath())));
 
-        return result.future();
+        return Future.succeededFuture();
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/controller/DownloadFileController.java:64]: same here, previous implementation succeed future only after sending file content to the response

File: src/main/java/com/epam/aidial/core/Proxy.java (hunk 1 above)
Comment (artsiomkorzun) [reply src/main/java/com/epam/aidial/core/Proxy.java:107]: fixed

File: src/main/java/com/epam/aidial/core/controller/DownloadFileController.java (hunk 3 above)
Comment (artsiomkorzun) [reply src/main/java/com/epam/aidial/core/controller/DownloadFileController.java:64]: Check the contract of handle.

**Review by Maxim-Gadalov — APPROVED** [review]

**Pull request merged**